src/Tasks/Microsoft.NET.Build.Tasks.UnitTests/GivenADependencyContextBuilder.cs

```diff
@@ -211,6 +213,168 @@ public void ItHandlesReferenceAndPackageReferenceNameCollisions()
+                new MockTaskItem($"/usr/Path/{reference}.dll", new Dictionary<string, string> {
+                    { "CopyLocal", "false" },
+                    { "FusionName", $"{reference}, Version=4.0.0.0, Culture=neutral, PublicKeyToken=null" },
+                    { "Version", "" },
+                })).ToArray();
+
+            ProjectContext projectContext = lockFile.CreateProjectContext(
+                FrameworkConstants.CommonFrameworks.NetCoreApp10.GetShortFolderName(),
+                runtime: null,
+                platformLibraryName: Constants.DefaultPlatformLibrary,
+                runtimeFrameworks: null,
+                isSelfContained: false);
+
+            Dictionary<string, SingleProjectInfo> referenceProjectInfos = new();
+
+            foreach (KeyValuePair<string, List<string>> kvp in dependencies)
+            {
+                projectContext.LockFileTarget.Libraries.Add(
+                    new LockFileTargetLibrary()
+                    {
+                        Name = kvp.Key,
+                        Version = new NuGetVersion(4, 0, 0),
+                        Type = withResources.Contains(kvp.Key) ? "project" : "unrealType",
+                        Dependencies = kvp.Value.Select(n => new PackageDependency(n)).ToList()
+                    });
+
+                if (withResources.Contains(kvp.Key))
+                {
+                    var fullPath = Path.GetFullPath(Path.Combine(mainProjectDirectory, kvp.Key));
+                    lockFile.Libraries.Add(new LockFileLibrary()
+                    {
+                        Name = kvp.Key,
+                        Version = new NuGetVersion(4, 0, 0),
+                        Type = "project",
+                        MSBuildProject = fullPath
+                    });
+
+                    referenceProjectInfos.Add(fullPath, SingleProjectInfo.Create(kvp.Key, kvp.Key, ".dll", "4.0.0",
+                        [new MockTaskItem($"{kvp.Key}.resource", new Dictionary<string, string>() {
+                            { "Culture", "en-us" },
+                            { "TargetPath", $"{kvp.Key}.resource" }
+                        })]));
+                }
+            }
+
+            CompilationOptions compilationOptions = CreateCompilationOptions();
+
+            return new DependencyContextBuilder(mainProject, includeRuntimeFileVersions: false, runtimeGraph: null, projectContext: projectContext, libraryLookup: new LockFileLookup(lockFile))
+                .WithReferenceAssemblies(ReferenceInfo.CreateReferenceInfos(referencePaths))
+                .WithCompilationOptions(compilationOptions)
+                .WithReferenceProjectInfos(referenceProjectInfos)
+                .Build();
+        }
+
+        [Fact]
+        public void DirectReferenceToPackageWithNoAssets()
+        {
+            DependencyContext dependencyContext = BuildDependencyContextFromDependenciesWithResources([], [], ["System.A"]);
```

Comment:
I'm not sure about how the references parameter is handled here... it looks like that's creating direct DLL references, and the tests may not model dependencies from the main project to a NuGet package, which is the more common scenario.

Reply:
Yes; it's from the main project to any of its direct references. I'm not sure I understand the distinction you're making between a direct reference from the main project to a dll vs. a direct reference to a nuget package? Are you saying that with the nuget packages we'd be downloading it just before using it? I don't see how that distinction is represented in DependencyContextBuilder, though I may be missing something.